Cloudinary/Shared/Coudinary.NetCoreShared/Url.cs

```diff
@@ -1,23 +1,136 @@
 ﻿using CloudinaryDotNet;
-using Coudinary.NetCoreShared;
-using Microsoft.AspNetCore.Html;
+using CloudinaryDotNet.Core;
 using System;
 using System.Collections.Generic;
 using System.Globalization;
-using System.Net;
+using System.Linq;
 using System.Text;
 using System.Text.RegularExpressions;
+using System.Threading.Tasks;
 
-namespace CloudinaryDotNet
+#if NET40
+using System.Web;
+#endif
+
+namespace CloudinaryShared.Core
 {
-    public class Url : UrlShared
+    public class Url : CloudinaryDotNet.Core.ICloneable
     {
-        public Url(string cloudName) : base(cloudName)
+        public static class Crc32
```

Comment:
can this class be extract to a different file? or make it private? (I didn't see it's used outside this class)